cw_monero/lib/monero_wallet.dart

```diff
@@ -266,6 +276,9 @@ abstract class MoneroWalletBase extends WalletBase<MoneroBalance,
     await walletAddresses.updateAddressesInBox();
     await backupWalletFiles(name);
     await monero_wallet.store();
+    if (isEnabledAutoGenerateSubaddress) {
+      walletAddresses.updateUnusedSubaddress(accountIndex: walletAddresses.account?.id ?? 0, defaultLabel: walletAddresses.account?.label ?? '');
+    }
```

Comment:
I think this should be called first before `updateAddressesInBox` right?
so that it saves the updated unused address

Reply:
You are right. I reworked the `updateAddressesInBox`-function to accommodate that logic. Because `updateUnusedSubaddress` requires the used subaddress to be uptodate.